### PR TITLE
Add arithmetic assignments handling

### PR DESCRIPTION
**Pull request by RRozak**
I added handling arithmetic assignments, such as `+=`.
It will fix that test: https://github.com/alainmarcel/uhdm-integration/pull/337/commits/86727dd7d2e8054352e4f4f08983d87ccbce9b61

### CODE REVIEW

File: src/UhdmAst.cpp [hunk 1]
```diff
@@ -854,10 +662,10 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
         return new AstOr(new FileLine("uhdm"), operands[0], operands[1]);
     }
     case vpiBitXorOp: {
-        return new AstXor(new FileLine("uhdm"), operands[1], operands[0]);
+        return new AstXor(new FileLine("uhdm"), operands[0], operands[1]);
     }
     case vpiBitXnorOp: {
-        return new AstXnor(new FileLine("uhdm"), operands[1], operands[0]);
+        return new AstXnor(new FileLine("uhdm"), operands[0], operands[1]);
```
Comment (rkapuscik) [inline src/UhdmAst.cpp:688]: Could you add a test case for those, preferably with asserts, to make sure we keep those indices right?

File: src/UhdmAst.cpp [hunk 2]
```diff
@@ -1103,6 +911,206 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
...
+    visit_one_to_one({vpiRhs}, obj_h, shared, [&](AstNode* childp) { rvaluep = childp; });
+
+    // Left
+    visit_one_to_one({vpiLhs}, obj_h, shared, [&](AstNode* childp) { lvaluep = childp; });
+
+    const unsigned int arithmeticAssignments[] = {vpiBitAndOp, vpiBitOrOp, vpiBitXorOp,
+                                                  vpiBitXnorOp, vpiLogAndOp, vpiLogOrOp,
+                                                  vpiSubOp, vpiAddOp, vpiMultOp, vpiDivOp,
+                                                  vpiModOp, vpiArithLShiftOp, vpiLShiftOp,
+                                                  vpiRShiftOp, vpiArithRShiftOp};
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: Variable name - those are not assignment types, but allowed operations. Maybe something along the lines of `arithmeticAssignmentOpTypes` would be better?

File: src/UhdmAst.cpp [hunk 3]
```diff
@@ -1103,6 +911,206 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
...
+    if (std::find(std::begin(arithmeticAssignments),
+                  std::end(arithmeticAssignments),
+                  operationType) != std::end(arithmeticAssignments)) {
+        rvaluep = process_operation(obj_h, shared, {lvaluep, rvaluep});
+        lvaluep = lvaluep->cloneTree(true);
+    }
+
+    if (rvaluep != nullptr && rvaluep->type() == AstType::en::atFOpen) {
+        // Not really an assignpment, AstFOpen node takes care of the lhs
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: Typo: assignPment

File: src/UhdmAst.cpp [hunk 4]
```diff
@@ -1103,6 +911,206 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
...
+            auto blocking = vpi_get(vpiBlocking, obj_h);
+            if (blocking) {
+                return new AstAssign(new FileLine("uhdm"), lvaluep, rvaluep);
+            } else {
+                return new AstAssignDly(new FileLine("uhdm"), lvaluep, rvaluep);
+            }
+        } else {
+            AstNode* assignp;
+            if (lvaluep->type() == AstType::en::atVar) {
+                // This is not a true assignpment
```
Comment (rkapuscik) [inline src/UhdmAst.cpp]: Typo: assignPment

**Review by rkapuscik — CHANGES_REQUESTED** [review]
Looks good, only a couple of cosmetic issues to be addressed.

File: src/UhdmAst.cpp (hunk 3 above)
Comment (RRozak) [reply src/UhdmAst.cpp]: It is copied from previous version of that function. I will correct it.

File: src/UhdmAst.cpp (hunk 2 above)
Comment (RRozak) [reply src/UhdmAst.cpp]: Yes, it would be better.
 First I named it arithmeticAndLogicalAssignmentOperations, and then tried to make it shorter.

**Review by rkapuscik — APPROVED** [review]

**Pull request merged**